File: DesignPatterns/singleton/singleton.ts
class Singleton {
  private static instance:Singleton
  private constructor() {
    
  }
  static getInstance(){
    if(!Singleton.instance){
      Singleton.instance = new Singleton()
    }
    return Singleton.instance
  }
  
}


const x = Singleton.getInstance()
const x2 = Singleton.getInstance()

console.log(x === x2) // true

File: src/Creational_Patterns/Singleton/examples/python/singleton.py
class SingletonMeta(type):
    _instances = {}
    def __call__(cls, *args, **kwargs):
        if cls not in cls._instances:
            instance = super().__call__(*args, **kwargs)
            cls._instances[cls] = instance
        return cls._instances[cls]


class Singleton(metaclass=SingletonMeta) :
  def getNumber(self):
    return 2


x = Singleton()
x2 = Singleton()

print(x2 == x)

